projects/gateway2/translator/sslutils/ssl_utils.go

```diff
@@ -59,3 +78,50 @@ func cleanedSslKeyPair(certChain, privateKey, rootCa string) (cleanedChain strin
 
 	return cleanedChain, err
 }
+
+// ApplySslExtensionOptions applies the GatewayTLSConfig options to the SslConfig
+// This function will never exit early, even if an error is encountered.
+// It will apply all options and return all errors encountered.
+func ApplySslExtensionOptions(ctx context.Context, in *gwv1.GatewayTLSConfig, out *ssl.SslConfig) error {
+	if len(in.Options) == 0 {
+		return nil
+	}
+
+	var err error
+
+	if oneWayTls, ok := in.Options[GatewaySslOneWayTls]; ok {
+		if strings.ToLower(string(oneWayTls)) == "true" {
+			out.OneWayTls = wrapperspb.Bool(true)
+		}
+	}
+
+	if varifySubjectAltNameStr, ok := in.Options[GatewaySslVerifySubjectAltName]; ok {
+		altNames := strings.Split(string(varifySubjectAltNameStr), ",")
+		out.VerifySubjectAltName = altNames
+	}
+
+	out.Parameters = &ssl.SslParameters{}
+	if cipherSuitesStr, ok := in.Options[GatewaySslCipherSuitesOption]; ok {
+		cipherSuites := strings.Split(string(cipherSuitesStr), ",")
+		out.Parameters.CipherSuites = cipherSuites
+	}
+
+	if minTlsVersion, ok := in.Options[GatewaySslMinimumTlsVersion]; ok {
+		if parsed, ok := ssl.SslParameters_ProtocolVersion_value[string(minTlsVersion)]; ok {
+			out.Parameters.MinimumProtocolVersion = ssl.SslParameters_ProtocolVersion(parsed)
+		} else {
+			contextutils.LoggerFrom(ctx).Debugf("invalid minimum tls version: %s", minTlsVersion)
+			err = multierror.Append(err, eris.Errorf("invalid minimum tls version: %s", minTlsVersion))
+		}
+	}
+
+	if maxTlsVersion, ok := in.Options[GatewaySslMaximumTlsVersion]; ok {
+		if parsed, ok := ssl.SslParameters_ProtocolVersion_value[string(maxTlsVersion)]; ok {
+			out.Parameters.MaximumProtocolVersion = ssl.SslParameters_ProtocolVersion(parsed)
+		} else {
+			contextutils.LoggerFrom(ctx).Debugf("invalid maximum tls version: %s", maxTlsVersion)
+			err = multierror.Append(err, eris.Errorf("invalid maximum tls version: %s", maxTlsVersion))
+		}
+	}
```

Comment:
What if we added a quick loop on the in.Options that checks if any of the entries are within a distance 2 Diffie–Hellman and not one of the known entries we log something?

That way we can find potential mistypings?

Reply:
This would be great but also seems like overkill to me? Personally I'd suggest just logging for any entry with a key we don't support and the user can figure out the typo. We could even list supported keys as part of the log message.